### PR TITLE
Feature/large messages multiple processor options

### PR DESCRIPTION
**Pull request by neckron**
Draft PR.
work in progress

### CODE REVIEW

File: src/main/java/com/avioconsulting/log4j/sqs/client/ConnectorClient.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			}
+
+		}
+	}
+
+	private void initSqsExtendedClient(ConnectorClientAttributes attributes) {
+		if(awsSQSExtendedClient == null) {
+			final ExtendedClientConfiguration extendedClientConfig =
+					new ExtendedClientConfiguration()
+							.withLargePayloadSupportEnabled(awsS3Client, attributes.getS3BucketName());
```
Comment (adammead) [inline src/main/java/com/avioconsulting/log4j/sqs/client/ConnectorClient.java]: `withLargePayloadSupportEnabled` is deprecated. Please use `withPayloadSupportEnabled` instead

File: src/main/java/com/avioconsulting/log4j/sqs/processor/S3MessageProcessor.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * use to send this message.
+ */
+public class S3MessageProcessor implements LogEventProcessor{
+
+	@Override public MessageRequestWrapper process(ProcessorAttributes processorAttributes) {
+		ObjectMetadata objectMetadata = new ObjectMetadata();
+		objectMetadata.setContentType("application/txt");
+		objectMetadata.setContentLength(processorAttributes.getMessage().length());
+		InputStream stream = new ByteArrayInputStream(processorAttributes.getMessage().getBytes(StandardCharsets.UTF_8));
+		PutObjectRequest putObjectRequest = new PutObjectRequest("sqs-extended-test-bucket", UUID.randomUUID().toString(), stream,objectMetadata);
```
Comment (adammead) [inline src/main/java/com/avioconsulting/log4j/sqs/processor/S3MessageProcessor.java]: Bucket name should be a variable here

File: src/main/java/com/avioconsulting/log4j/sqs/SqsManager.java [hunk 3]
```diff
@@ -80,112 +53,27 @@ public void startup() {
...
-            ss.add(s);                                  // add what has been encoded to the list
-            pos = newpos;                               // store new input position
-            out.rewind();                               // and rewind output buffer
-            if (!result.isOverflow()) {
-                break;                                  // everything has been encoded
-            }
-        }
-        return ss.toArray(new String[0]);
+    private void sendMessageList(String processorType, String message, String queueUrl, Integer maxMessageBytes) {
+        LogEventProcessor processor = ProcessorSupplier.selectProcessor(processorType);
```
Comment (adammead) [inline src/main/java/com/avioconsulting/log4j/sqs/SqsManager.java]: Would it be better to select the message processor at the beginning and then use that for everything? That way we're not loading the same class repeatedly.

**Review by adammead — CHANGES_REQUESTED** [review]
A few inline comments. There are a couple suggestions.

File: src/main/java/com/avioconsulting/log4j/sqs/processor/S3MessageProcessor.java (hunk 2 above)
Comment (neckron) [reply src/main/java/com/avioconsulting/log4j/sqs/processor/S3MessageProcessor.java]: Got it.  changes in next commit.

File: src/main/java/com/avioconsulting/log4j/sqs/client/ConnectorClient.java (hunk 1 above)
Comment (neckron) [reply src/main/java/com/avioconsulting/log4j/sqs/client/ConnectorClient.java]: Got it.  changes in next commit.

File: src/main/java/com/avioconsulting/log4j/sqs/SqsManager.java (hunk 3 above)
Comment (neckron) [reply src/main/java/com/avioconsulting/log4j/sqs/SqsManager.java]: makes sense, let me check how I can improve that. Thanks for the suggestion.

Comment (neckron) [reply src/main/java/com/avioconsulting/log4j/sqs/SqsManager.java]: Did some improvements. However ProcessorSupplier has all the processors already instantiated ,so, depending on 'largeMessageMode' param,  supplier will return unique instance for each case. In scenario when message size < maxMessageBytes , Default processor is inyected, therefore, if it's needed, supplier has the capacity to retrieve other processor type depending on the situation (I added it to support further enhancements).

**Review by adammead — APPROVED** [review]

**Review by adesjardin — APPROVED** [review]

**Pull request merged**